src/bot.js

```diff
@@ -165,6 +165,46 @@ function addLFG(MESSAGE) {
         });
 }
 
+function endSession(message){
+    var author = message.author,
+        guild_id = message.guild.id,
+        role_id = '',
+        roles = message.member.roles.array(),
+        role = null;
+
+    for(var i = 0; i < roles.length; i++){
+        role = roles[i];
+
+        if(role.name.startsWith('lfg_')){
+            role_id = role.id;
+        }
+    }
+
+    if(guild_id != null && (role_id != null || role_id != '')){
+        config.removeSession(guild_id, role_id);
+        role.delete();
+
+        var channels = message.guild.channels.array();
+
+        for(var x = 0; x < channels.length; x++){
+            var channel = channels[x];
+
+            if(channel.name === role.name){
+                channel.delete()
+                    .then()
+                    .catch(console.error);
+            }
+        }
+
+        message.reply('Session has ended');
+    } else {
+        message.reply('Oops! Something went wrong');
```

Comment:
This code also runs if the user doesn't have an lfg role. They may not notice it (never trust the user) so I'd suggest adding a message for that.